<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ul.lst-kix_gkuq4v9q0p0z-4{list-style-type:none}ul.lst-kix_2ebh0drh04i4-0{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-3{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-2{list-style-type:none}ul.lst-kix_2ebh0drh04i4-2{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-1{list-style-type:none}ul.lst-kix_2ebh0drh04i4-1{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-0{list-style-type:none}ul.lst-kix_2ebh0drh04i4-4{list-style-type:none}ul.lst-kix_2ebh0drh04i4-3{list-style-type:none}ul.lst-kix_2ebh0drh04i4-6{list-style-type:none}ul.lst-kix_2ebh0drh04i4-5{list-style-type:none}ul.lst-kix_2ebh0drh04i4-8{list-style-type:none}ul.lst-kix_2ebh0drh04i4-7{list-style-type:none}.lst-kix_gkuq4v9q0p0z-8>li:before{content:"\0025a0  "}ul.lst-kix_gkuq4v9q0p0z-8{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-7{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-6{list-style-type:none}ul.lst-kix_gkuq4v9q0p0z-5{list-style-type:none}.lst-kix_gkuq4v9q0p0z-6>li:before{content:"\0025cf  "}.lst-kix_gkuq4v9q0p0z-7>li:before{content:"\0025cb  "}.lst-kix_gkuq4v9q0p0z-3>li:before{content:"\0025cf  "}.lst-kix_2ebh0drh04i4-0>li:before{content:"\0025cf  "}.lst-kix_gkuq4v9q0p0z-4>li:before{content:"\0025cb  "}.lst-kix_gkuq4v9q0p0z-5>li:before{content:"\0025a0  "}.lst-kix_2ebh0drh04i4-1>li:before{content:"\0025cb  "}.lst-kix_2ebh0drh04i4-4>li:before{content:"\0025cb  "}.lst-kix_2ebh0drh04i4-2>li:before{content:"\0025a0  "}.lst-kix_2ebh0drh04i4-6>li:before{content:"\0025cf  "}.lst-kix_2ebh0drh04i4-3>li:before{content:"\0025cf  "}.lst-kix_2ebh0drh04i4-7>li:before{content:"\0025cb  "}.lst-kix_2ebh0drh04i4-5>li:before{content:"\0025a0  "}.lst-kix_2ebh0drh04i4-8>li:before{content:"\0025a0  "}.lst-kix_tdr03w5igcwc-1>li:before{content:"\0025cb  "}.lst-kix_tdr03w5igcwc-0>li:before{content:"\0025cf  "}.lst-kix_tdr03w5igcwc-2>li:before{content:"\0025a0  "}.lst-kix_tdr03w5igcwc-3>li:before{content:"\0025cf  "}.lst-kix_tdr03w5igcwc-5>li:before{content:"\0025a0  "}.lst-kix_tdr03w5igcwc-4>li:before{content:"\0025cb  "}.lst-kix_tdr03w5igcwc-7>li:before{content:"\0025cb  "}.lst-kix_tdr03w5igcwc-6>li:before{content:"\0025cf  "}.lst-kix_tdr03w5igcwc-8>li:before{content:"\0025a0  "}ul.lst-kix_tdr03w5igcwc-7{list-style-type:none}ul.lst-kix_tdr03w5igcwc-8{list-style-type:none}.lst-kix_gkuq4v9q0p0z-0>li:before{content:"\0025cf  "}ul.lst-kix_tdr03w5igcwc-5{list-style-type:none}ul.lst-kix_tdr03w5igcwc-6{list-style-type:none}ul.lst-kix_tdr03w5igcwc-3{list-style-type:none}ul.lst-kix_tdr03w5igcwc-4{list-style-type:none}.lst-kix_gkuq4v9q0p0z-2>li:before{content:"\0025a0  "}.lst-kix_gkuq4v9q0p0z-1>li:before{content:"\0025cb  "}ul.lst-kix_tdr03w5igcwc-1{list-style-type:none}ul.lst-kix_tdr03w5igcwc-2{list-style-type:none}ul.lst-kix_tdr03w5igcwc-0{list-style-type:none}ol{margin:0;padding:0}.c1{orphans:2;widows:2;direction:ltr;height:11pt}.c3{padding-top:24pt;orphans:2;widows:2;direction:ltr}.c5{background-color:#ffffff;max-width:468pt;padding:72pt 72pt 72pt 72pt}.c4{orphans:2;widows:2;direction:ltr}.c11{color:inherit;text-decoration:inherit}.c10{color:#1155cc;text-decoration:underline}.c6{padding:0;margin:0}.c8{margin-left:72pt;padding-left:0pt}.c12{font-weight:bold}.c9{padding-left:0pt}.c14{font-size:8pt}.c7{page-break-after:avoid}.c2{margin-left:36pt}.c0{font-family:"Courier New"}.c13{padding-top:24pt}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c5"><p class="c1 c13 subtitle"><a name="h.ky013g3qen1k"></a></p><p class="c4 c7 title"><a name="h.dm6m5hjs959h"></a><span>How does the analysis of Amplicon metagenomes and 16S data extracted from WGS metagenomes compare?</span></p><p class="c3 subtitle"><a name="h.nb8y5lwmnusu"></a><span>Cross-biome metagenomic analyses of soil microbial communities and their functional attributes</span></p><p class="c4"><span class="c12 c14">Fierer et al. (doi: 10.1073/pnas.1215210110 ; </span><span class="c10 c12 c14"><a class="c11" href="https://www.google.com/url?q=http://www.pnas.org/content/109/52/21390.full&amp;sa=D&amp;usg=AFQjCNGJ6Ak0dZae0cLXe6dDGM6bKkNqeA">http://www.pnas.org/content/109/52/21390.full</a></span><span class="c12 c14">)</span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><p class="c4"><span class="c12">Samples in MG-RAST:</span></p><ul class="c6 lst-kix_tdr03w5igcwc-0 start"><li class="c4 c2 c9"><span>16S: &nbsp;(</span><span class="c10"><a class="c11" href="https://www.google.com/url?q=http://metagenomics.anl.gov/metagenomics.cgi?page%3DMetagenomeProject%26project%3D10307&amp;sa=D&amp;usg=AFQjCNG57RYJtRHNTPQVwC6c6Y2mdRIeaQ">http://metagenomics.anl.gov/metagenomics.cgi?page=MetagenomeProject&amp;project=10307</a></span><span>) <br>mgm4575333.3,mgm4575334.3,mgm4575335.3,mgm4575336.3,mgm4575337.3,mgm4575338.3,mgm4575339.3,mgm4575340.3,mgm4575341.3,mgm4575342.3,mgm4575343.3,mgm4575344.3,mgm4575345.3,mgm4575346.3,mgm4575347.3,mgm4575348.3</span></li><li class="c4 c9 c2"><span>WGS: (</span><span class="c10"><a class="c11" href="https://www.google.com/url?q=http://metagenomics.anl.gov/metagenomics.cgi?page%3DMetagenomeProject%26project%3D2997&amp;sa=D&amp;usg=AFQjCNGOMmmA8ivAFHX8hR2Z-YlaQ3QS4A">http://metagenomics.anl.gov/metagenomics.cgi?page=MetagenomeProject&amp;project=2997</a></span><span>) <br>mgm4477904.3,mgm4477903.3,mgm4477902.3,mgm4477901.3,mgm4477900.3,mgm4477899.3,mgm4477877.3,mgm4477876.3,mgm4477875.3,mgm4477874.3,mgm4477873.3,mgm4477872.3,mgm4477807.3,mgm4477805.3,mgm4477804.3,mgm4477803.3</span></li></ul><p class="c1"><span></span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><h2 class="c4 c7"><a name="h.nl2vfpkj4395"></a><span>Scripts</span></h2><p class="c4"><span>We will use scripts from the MG-RAST-Tools repository on GitHub (</span><span class="c10"><a class="c11" href="https://www.google.com/url?q=https://github.com/MG-RAST/MG-RAST-Tools&amp;sa=D&amp;usg=AFQjCNHECMTBlDiJEe4NJPhVp7eM10X5nw">https://github.com/MG-RAST/MG-RAST-Tools</a></span><span>). Feel free to copy, edit or contribute.</span></p><p class="c1"><span></span></p><p class="c1"><span></span></p><ul class="c6 lst-kix_gkuq4v9q0p0z-0 start"><li class="c4 c9 c2"><span>mg-compare-taxa.py</span></li><li class="c4 c9 c2"><span>mg-mg-compare-normalize.py</span></li><li class="c4 c9 c2"><span>mg-biom-merge.py</span></li><li class="c4 c9 c2"><span>mg-biom2metadata</span></li><li class="c4 c9 c2"><span>mg-biom2taxa</span></li></ul><p class="c1"><span></span></p><h2 class="c4 c7"><a name="h.7z8wos8lxoy3"></a><span>Workflow</span></h2><p class="c1"><span></span></p><p class="c4"><span>We have to create abundance profiles for both sample groups and import them into R to compare them with phyloseq. </span></p><p class="c1"><span></span></p><ul class="c6 lst-kix_2ebh0drh04i4-0 start"><li class="c4 c9 c2"><span>Create Greengenes abundance profiles for both sample groups ( Amplicon &nbsp;and 16S from WGS ).<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-1 start"><li class="c4 c8"><span>mg-compare-taxa.py --ids $WGS --level genus --source Greengenes --format biom --evalue 8 --hit_type all &gt; wgs.greengenes.biom</span></li><li class="c4 c8"><span>mg-compare-taxa.py --ids $AMPLICON --level genus --source Greengenes --format biom --evalue 8 --hit_type all &gt; amplicon.greengenes.biom<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-0"><li class="c4 c9 c2"><span>We have to combine the two profiles.<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-1 start"><li class="c4 c8"><span>mg-biom-merge.py wgs.greengenes.biom amplicon.greengenes.biom &gt; merged.greengenes.biom<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-0"><li class="c4 c9 c2"><span>Normalizing the profiles, this can be done in R too.<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-1 start"><li class="c4 c8"><span>mg-mg-compare-normalize.py --input merged.greengenes.biom --output merged.greengenes.normalized.biom<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-0"><li class="c4 c9 c2"><span>The OTU data from the biom file, metadata and taxonomy information are needed to create a phyloseq object [pyholoseq_object = phyloseq(OTUTABLE, TAX , META)]. For this we extract the metadata and taxonomy information from the biom file.</span></li></ul><p class="c1 c2"><span></span></p><p class="c1"><span></span></p><ul class="c6 lst-kix_2ebh0drh04i4-1"><li class="c4 c8"><span>mg-biom2metadata --input amplicon.greengenes.biom &gt; amplicon.metadata</span></li><li class="c4 c8"><span>mg-biom2taxa --input amplicon.greengenes.biom &gt; amplicon.taxa<br><br>Same for wgs.greengenes.biom and merged.greengenes.biom<br></span></li></ul><ul class="c6 lst-kix_2ebh0drh04i4-0"><li class="c4 c9 c2"><span>Switch to R for further analysis.<br></span></li></ul><p class="c1"><span></span></p><p class="c1"><span></span></p><h1 class="c4 c7"><a name="h.s9al3jcqa9uu"></a><span>R Script</span></h1><p class="c1"><span></span></p><p class="c4"><span class="c0">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span></p><p class="c4"><span class="c0">biocLite(&quot;phyloseq&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">install.packages(&quot;ape&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;ade4&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;doParallel&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;foreach&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;ggplot2&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;igraph0&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;picante&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;plyr&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;RJSONIO&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;scales&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;testthat&quot;)</span></p><p class="c4"><span class="c0">install.packages(&quot;vegan&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">library(ggplot2)</span></p><p class="c4"><span class="c0">library(&quot;biom&quot;)</span></p><p class="c4"><span class="c0">library(&quot;phyloseq&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># reading json/biom file</span></p><p class="c4"><span class="c0">amplicon.biom = read_biom(&quot;/Users/Andi/Development/tmp/data/phyloseq/amplicon.greengenes.biom&quot;)</span></p><p class="c4"><span class="c0">wgs.biom &nbsp; &nbsp; &nbsp;= read_biom(&quot;/Users/Andi/Development/tmp/data/phyloseq/wgs.greengenes.biom&quot;)</span></p><p class="c4"><span class="c0">merged.biom &nbsp; &nbsp; &nbsp;= read_biom(&quot;/Users/Andi/Development/tmp/data/phyloseq/merged.greengenes.biom&quot;)</span></p><p class="c4"><span class="c0">#refseq &nbsp; &nbsp;= read_biom(&quot;~/Development/tmp/data/phyloseq/test4.biom&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">### Create amplicon object</span></p><p class="c4"><span class="c0"># OTU matrix</span></p><p class="c4"><span class="c0">otumatrix = as(biom_data(amplicon.biom), &quot;matrix&quot;)</span></p><p class="c4"><span class="c0">OTU = otu_table(otumatrix, taxa_are_rows=TRUE)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># import metadata</span></p><p class="c4"><span class="c0">metadata_file=&quot;/Users/Andi/Development/tmp/data/phyloseq/amplicon.metadata&quot;</span></p><p class="c4"><span class="c0">META = import_qiime_sample_data(metadata_file)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># import taxa</span></p><p class="c4"><span class="c0">taxa_file=&quot;/Users/Andi/Development/tmp/data/phyloseq/amplicon.taxa&quot;</span></p><p class="c4"><span class="c0">taxa.frame &lt;- read.csv(taxa_file , header = TRUE , sep = &quot;\t&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">taxmat = as.matrix(taxa.frame)</span></p><p class="c4"><span class="c0">rownames(taxmat) &lt;- taxmat[,1]</span></p><p class="c4"><span class="c0">taxmat &lt;- taxmat[,-1]</span></p><p class="c4"><span class="c0">#colnames(taxmat2) &lt;- c(&quot;Domain&quot;, &quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, </span></p><p class="c4"><span class="c0"># &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&quot;Species&quot;)</span></p><p class="c4"><span class="c0">TAX = tax_table(taxmat)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">amplicon = phyloseq(OTU, TAX , META)</span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">### Create wgs object</span></p><p class="c4"><span class="c0"># OTU matrix</span></p><p class="c4"><span class="c0">otumatrix = as(biom_data(wgs.biom), &quot;matrix&quot;)</span></p><p class="c4"><span class="c0">OTU = otu_table(otumatrix, taxa_are_rows=TRUE)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># import metadata</span></p><p class="c4"><span class="c0">metadata_file=&quot;/Users/Andi/Development/tmp/data/phyloseq/wgs.metadata&quot;</span></p><p class="c4"><span class="c0">META = import_qiime_sample_data(metadata_file)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># import taxa</span></p><p class="c4"><span class="c0">taxa_file=&quot;/Users/Andi/Development/tmp/data/phyloseq/wgs.taxa&quot;</span></p><p class="c4"><span class="c0">taxa.frame &lt;- read.csv(taxa_file , header = TRUE , sep = &quot;\t&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">taxmat = as.matrix(taxa.frame)</span></p><p class="c4"><span class="c0">rownames(taxmat) &lt;- taxmat[,1]</span></p><p class="c4"><span class="c0">taxmat &lt;- taxmat[,-1]</span></p><p class="c4"><span class="c0">#colnames(taxmat2) &lt;- c(&quot;Domain&quot;, &quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, </span></p><p class="c4"><span class="c0"># &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&quot;Species&quot;)</span></p><p class="c4"><span class="c0">TAX = tax_table(taxmat)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">wgs = phyloseq(OTU, TAX , META)</span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">### Create merged 16S object</span></p><p class="c4"><span class="c0"># OTU matrix</span></p><p class="c4"><span class="c0">otumatrix = as(biom_data(merged.biom), &quot;matrix&quot;)</span></p><p class="c4"><span class="c0">OTU = otu_table(otumatrix, taxa_are_rows=TRUE)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># import metadata</span></p><p class="c4"><span class="c0">metadata_file=&quot;/Users/Andi/Development/tmp/data/phyloseq/merged.metadata&quot;</span></p><p class="c4"><span class="c0">META = import_qiime_sample_data(metadata_file)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># import taxa</span></p><p class="c4"><span class="c0">taxa_file=&quot;/Users/Andi/Development/tmp/data/phyloseq/merged.taxa&quot;</span></p><p class="c4"><span class="c0">taxa.frame &lt;- read.csv(taxa_file , header = TRUE , sep = &quot;\t&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">taxmat = as.matrix(taxa.frame)</span></p><p class="c4"><span class="c0">rownames(taxmat) &lt;- taxmat[,1]</span></p><p class="c4"><span class="c0">taxmat &lt;- taxmat[,-1]</span></p><p class="c4"><span class="c0">#colnames(taxmat2) &lt;- c(&quot;Domain&quot;, &quot;Phylum&quot;, &quot;Class&quot;, &quot;Order&quot;, &quot;Family&quot;, &quot;Genus&quot;, </span></p><p class="c4"><span class="c0"># &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;&quot;Species&quot;)</span></p><p class="c4"><span class="c0">TAX = tax_table(taxmat)</span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0">merged = phyloseq(OTU, TAX , META)</span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># Plots Alpha Diversity Amplicon</span></p><p class="c4"><span class="c0">plot_richness(amplicon) + geom_boxplot()</span></p><p class="c4"><span class="c0">plot_ordination(amplicon , ordinate(amplicon) ) + geom_point(size=10)</span></p><p class="c4"><span class="c0">plot_ordination(amplicon , ordinate(amplicon, &quot;MDS&quot;) ) + geom_point(size=10)</span></p><p class="c4"><span class="c0">plot_ordination(amplicon , ordinate(amplicon, &quot;PCoA&quot;) , color = &quot;sample.data.biome&quot;) + geom_point(size=5)</span></p><p class="c4"><span class="c0">plot_heatmap(amplicon , title = &quot;Cross-Site&quot; , taxa.label = &quot;Genus&quot;)</span></p><p class="c4"><span class="c0">plot_bar(amplicon, fill = &quot;Family&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># Plots Alpha Diversity WGS</span></p><p class="c4"><span class="c0">plot_richness(wgs) + geom_boxplot()</span></p><p class="c4"><span class="c0">plot_ordination(wgs , ordinate(wgs) ) + geom_point(size=10)</span></p><p class="c4"><span class="c0">plot_ordination(wgs , ordinate(wgs, &quot;MDS&quot;) ) + geom_point(size=10)</span></p><p class="c4"><span class="c0">plot_ordination(wgs , ordinate(wgs, &quot;PCoA&quot;) , color = &quot;sample.data.biome&quot;) + geom_point(size=5)</span></p><p class="c4"><span class="c0">plot_heatmap(wgs , title = &quot;Cross-Site (WGS)&quot; , taxa.label = &quot;Genus&quot;)</span></p><p class="c4"><span class="c0">plot_bar(wgs, fill = &quot;Family&quot;)</span></p><p class="c1"><span class="c0"></span></p><p class="c1"><span class="c0"></span></p><p class="c4"><span class="c0"># Plots Alpha Diversity Merged</span></p><p class="c4"><span class="c0">plot_richness(merged) + geom_boxplot()</span></p><p class="c4"><span class="c0">plot_ordination(merged , ordinate(merged) ) + geom_point(size=10)</span></p><p class="c4"><span class="c0">plot_ordination(merged , ordinate(merged, &quot;MDS&quot;) ) + geom_point(size=10)</span></p><p class="c4"><span class="c0">plot_ordination(merged , ordinate(merged, &quot;PCoA&quot;) , color = &quot;sample.data.biome&quot;) + geom_point(size=5)</span></p><p class="c4"><span class="c0">plot_heatmap(merged , title = &quot;Cross-Site (WGS)&quot; , taxa.label = &quot;Genus&quot;)</span></p><p class="c4"><span class="c0">plot_bar(merged, fill = &quot;Family&quot;)</span></p><p class="c1"><span></span></p><p class="c4"><span class="c0">plot_ordination(wgs , ordinate(wgs, &quot;PCoA&quot;) , color = &quot;library.type&quot;) + geom_point(size=5)</span></p><p class="c4"><span class="c0">plot_ordination(amplicon , ordinate(amplicon, &quot;PCoA&quot;) , color = &quot;library.type&quot;) + geom_point(size=5)</span></p><p class="c4"><span class="c0">plot_ordination(merged , ordinate(merged, &quot;PCoA&quot;) , color = &quot;library.type&quot;) + geom_point(size=5)</span></p><p class="c1"><span></span></p><p class="c1"><span></span></p></body></html>