<html><head><meta content="text/html; charset=UTF-8" http-equiv="content-type"><style type="text/css">ul.lst-kix_2ebh0drh04i4-0{list-style-type:none}ul.lst-kix_2ebh0drh04i4-2{list-style-type:none}ul.lst-kix_2ebh0drh04i4-4{list-style-type:none}ul.lst-kix_2ebh0drh04i4-3{list-style-type:none}.lst-kix_tdr03w5igcwc-2>li:before{content:"\0025a0  "}ul.lst-kix_2ebh0drh04i4-6{list-style-type:none}ul.lst-kix_2ebh0drh04i4-5{list-style-type:none}.lst-kix_tdr03w5igcwc-3>li:before{content:"\0025cf  "}ul.lst-kix_2ebh0drh04i4-8{list-style-type:none}ul.lst-kix_2ebh0drh04i4-7{list-style-type:none}.lst-kix_tdr03w5igcwc-5>li:before{content:"\0025a0  "}.lst-kix_tdr03w5igcwc-4>li:before{content:"\0025cb  "}.lst-kix_tdr03w5igcwc-7>li:before{content:"\0025cb  "}.lst-kix_tdr03w5igcwc-6>li:before{content:"\0025cf  "}.lst-kix_2ebh0drh04i4-0>li:before{content:"\0025cf  "}.lst-kix_tdr03w5igcwc-8>li:before{content:"\0025a0  "}.lst-kix_2ebh0drh04i4-1>li:before{content:"" counter(lst-ctn-kix_2ebh0drh04i4-1,lower-latin) ". "}.lst-kix_2ebh0drh04i4-4>li:before{content:"\0025cb  "}ul.lst-kix_tdr03w5igcwc-7{list-style-type:none}ul.lst-kix_tdr03w5igcwc-8{list-style-type:none}ol.lst-kix_2ebh0drh04i4-1{list-style-type:none}ul.lst-kix_tdr03w5igcwc-5{list-style-type:none}.lst-kix_2ebh0drh04i4-2>li:before{content:"\0025a0  "}.lst-kix_2ebh0drh04i4-6>li:before{content:"\0025cf  "}ul.lst-kix_tdr03w5igcwc-6{list-style-type:none}ul.lst-kix_tdr03w5igcwc-3{list-style-type:none}.lst-kix_2ebh0drh04i4-3>li:before{content:"\0025cf  "}.lst-kix_2ebh0drh04i4-7>li:before{content:"\0025cb  "}ul.lst-kix_tdr03w5igcwc-4{list-style-type:none}.lst-kix_2ebh0drh04i4-1>li{counter-increment:lst-ctn-kix_2ebh0drh04i4-1}.lst-kix_2ebh0drh04i4-5>li:before{content:"\0025a0  "}ul.lst-kix_tdr03w5igcwc-1{list-style-type:none}.lst-kix_2ebh0drh04i4-8>li:before{content:"\0025a0  "}ul.lst-kix_tdr03w5igcwc-2{list-style-type:none}ol.lst-kix_2ebh0drh04i4-1.start{counter-reset:lst-ctn-kix_2ebh0drh04i4-1 0}ul.lst-kix_tdr03w5igcwc-0{list-style-type:none}.lst-kix_tdr03w5igcwc-1>li:before{content:"\0025cb  "}.lst-kix_tdr03w5igcwc-0>li:before{content:"\0025cf  "}ol{margin:0;padding:0}.c2{padding-top:24pt;orphans:2;widows:2;direction:ltr;height:11pt}.c3{orphans:2;widows:2;direction:ltr;height:11pt}.c10{background-color:#ffffff;max-width:468pt;padding:72pt 72pt 72pt 72pt}.c5{orphans:2;widows:2;direction:ltr}.c9{color:#1155cc;text-decoration:underline}.c1{color:inherit;text-decoration:inherit}.c11{padding:0;margin:0}.c7{margin-left:36pt;padding-left:0pt}.c0{font-size:10pt;font-family:"Courier New"}.c4{font-family:"Courier New"}.c6{padding-top:24pt}.c8{page-break-after:avoid}.title{padding-top:0pt;color:#000000;font-size:26pt;padding-bottom:3pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}.subtitle{padding-top:0pt;color:#666666;font-size:15pt;padding-bottom:16pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}li{color:#000000;font-size:11pt;font-family:"Arial"}p{margin:0;color:#000000;font-size:11pt;font-family:"Arial"}h1{padding-top:20pt;color:#000000;font-size:20pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h2{padding-top:18pt;color:#000000;font-size:16pt;padding-bottom:6pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h3{padding-top:16pt;color:#434343;font-size:14pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h4{padding-top:14pt;color:#666666;font-size:12pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h5{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;orphans:2;widows:2;text-align:left}h6{padding-top:12pt;color:#666666;font-size:11pt;padding-bottom:4pt;font-family:"Arial";line-height:1.15;page-break-after:avoid;font-style:italic;orphans:2;widows:2;text-align:left}</style></head><body class="c10"><p class="c2 subtitle"><a name="h.651rt1p6is4q"></a></p><p class="c5"><span>This example shows the extraction of abundance-table data from a handful of shotgun samples and a handful of 16S samples from the MG-RAST API on the command line and the production of an ordination graph and comparative richness graphs using the phyloseq package in R.</span></p><p class="c3"><span></span></p><p class="c5"><span>The scripts used below (mg-biom-merge.py, mg-compare-taxa.py) are available on GitHub (</span><span class="c9"><a class="c1" href="https://www.google.com/url?q=https://github.com/MG-RAST/MG-RAST-Tools&amp;sa=D&amp;usg=AFQjCNHECMTBlDiJEe4NJPhVp7eM10X5nw">https://github.com/MG-RAST/MG-RAST-Tools</a></span><span>).</span></p><p class="c2 subtitle"><a name="h.knafcjdndq5v"></a></p><p class="c5 c6 subtitle"><a name="h.dcv0v0b4cuo"></a><span>Data from:</span></p><p class="c5 c6"><span>Cross-biome metagenomic analys</span><span>es of soil microbial communities and their functional attributes</span></p><p class="c5"><span>Fierer et al. (doi: 10.1073/pnas.1215210110 ; </span><span class="c9"><a class="c1" href="https://www.google.com/url?q=http://www.pnas.org/content/109/52/21390.full&amp;sa=D&amp;usg=AFQjCNGJ6Ak0dZae0cLXe6dDGM6bKkNqeA">http://www.pnas.org/content/109/52/21390.full</a></span><span>) MR-RAST project </span><span class="c9"><a class="c1" href="https://www.google.com/url?q=http://metagenomics.anl.gov/metagenomics.cgi?page%3DMetagenomeProject%26project%3D2997&amp;sa=D&amp;usg=AFQjCNGOMmmA8ivAFHX8hR2Z-YlaQ3QS4A">http://metagenomics.anl.gov/metagenomics.cgi?page=MetagenomeProject&amp;project=2997</a></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c5"><span>Workflow:</span></p><p class="c3"><span></span></p><ul class="c11 lst-kix_2ebh0drh04i4-0 start"><li class="c5 c7"><span>Compute &nbsp;abundance profiles for both sample sets and create a unified set for the analysis in R (shell script)</span></li><li class="c5 c7"><span>Import data in R, create phyloseq object and perform analysis (R script).</span></li></ul><p class="c3"><span></span></p><h2 class="c5 c8"><a name="h.xaswl0hanp55"></a><span>Shell script</span></h2><p class="c3"><span></span></p><p class="c5"><span class="c4"># Define list of sample IDs for both data sets</span></p><p class="c5"><span class="c4">WGS=mgm4477904.3,mgm4477903.3,mgm4477902.3,mgm4477901.3,mgm4477900.3,mgm4477899.3,mgm4477877.3,mgm4477876.3,mgm4477875.3,mgm4477874.3,mgm4477873.3,mgm4477872.3,mgm4477807.3,mgm4477805.3,mgm4477804.3,mgm4477803.3</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4">AMPLICON=mgm4575333.3,mgm4575334.3,mgm4575335.3,mgm4575336.3,mgm4575337.3,mgm4575338.3,mgm4575339.3,mgm4575340.3,mgm4575341.3,mgm4575342.3,mgm4575343.3,mgm4575344.3,mgm4575345.3,mgm4575346.3,mgm4575347.3,mgm4575348.3</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># Query MG-RAST for table of genus-level annotations based on similarity to 16S sequences &nbsp;for these 16 shotgun samples.</span></p><p class="c5"><span class="c4">mg-compare-taxa.py --ids $WGS --level genus --source Greengenes --format biom --evalue 8 --hit_type all &gt; wgs.greengenes.biom</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># And also get genus-level annotations from the 16S samples (using the same database)</span></p><p class="c5"><span class="c4">mg-compare-taxa.py --ids $AMPLICON --level genus --source Greengenes --format biom --evalue 8 --hit_type all &gt; amplicon.greengenes.biom</span></p><p class="c5"><span class="c4"># Combine the two tables into one biom object</span></p><p class="c5"><span class="c4">mg-biom-merge.py &nbsp;amplicon.greengenes.biom wgs.greengenes.biom &gt; merged.greengenes.biom</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># extract metadata and taxon tables from the biom files (required by phyloseq)</span></p><p class="c5"><span class="c4">mg-biom2metadata --input merged.greengenes.biom &gt; merged.metadata</span></p><p class="c5"><span class="c4">mg-biom2taxa --input merged.greengenes.biom &gt; merged.taxa</span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span><br></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><h1 class="c5 c8"><a name="h.s9al3jcqa9uu"></a><span>R Script</span></h1><p class="c3"><span></span></p><p class="c5"><span class="c4">source(&quot;http://bioconductor.org/biocLite.R&quot;)</span></p><p class="c5"><span class="c4">biocLite(&quot;phyloseq&quot;)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4">install.packages(&quot;ape&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;ade4&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;doParallel&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;foreach&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;ggplot2&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;igraph0&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;picante&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;plyr&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;RJSONIO&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;scales&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;testthat&quot;)</span></p><p class="c5"><span class="c4">install.packages(&quot;vegan&quot;)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4">library(ggplot2)</span></p><p class="c5"><span class="c4">library(&quot;biom&quot;)</span></p><p class="c5"><span class="c4">library(&quot;phyloseq&quot;)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># Import data from biom file</span></p><p class="c5"><span class="c4">PATH= # PATH_TO_MERGED_BIOM_FILE</span></p><p class="c5"><span class="c4">merged.biom = read_biom( paste( PATH , &ldquo;merged.greengenes.biom&rdquo; , sep=&quot;&quot;))</span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4">### Create phyloseq object from merged data sets</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># OTU matrix</span></p><p class="c5"><span class="c4">otumatrix = as(biom_data(amplicon.biom), &quot;matrix&quot;)</span></p><p class="c5"><span class="c4">OTU = otu_table(otumatrix, taxa_are_rows=TRUE)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># Import metadata</span></p><p class="c5"><span class="c4">metadata_file=paste(PATH , &rdquo;merged.metadata&quot; , sep=&rdquo;&rdquo;)</span></p><p class="c5"><span class="c4">META = import_qiime_sample_data(metadata_file)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># import taxa</span></p><p class="c5"><span class="c4">taxa_file=paste(PATH , &rdquo;merged.taxa&quot; , sep=&rdquo;&rdquo;)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4">taxa.frame &lt;- read.csv(taxa_file , header = TRUE , sep = &quot;\t&quot;)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4">taxmat = as.matrix(taxa.frame)</span></p><p class="c5"><span class="c4">rownames(taxmat) &lt;- taxmat[,1]</span></p><p class="c5"><span class="c4">taxmat &lt;- taxmat[,-1]</span></p><p class="c5"><span class="c4">TAX = tax_table(taxmat)</span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># pyloseq object from otutable , metadata and taxonomy information</span></p><p class="c5"><span class="c4">merged = phyloseq(OTU, TAX , META)</span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># Plot Alpha Diversity Merged</span></p><p class="c5"><span class="c4">plot_richness(merged) + geom_boxplot()</span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 542.67px;"><img alt="Merged-Alpha-Diversity-Raw.png" src="images/image01.png" style="width: 624.00px; height: 542.67px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span class="c4"></span></p><p class="c3"><span class="c4"></span></p><p class="c5"><span class="c4"># Ordination plot </span></p><p class="c5"><span class="c4">plot_ordination(merged , ordinate(merged, &quot;PCoA&quot;) , color = &quot;sample.data.biome&quot; , shape = &quot;library.type&quot;) + geom_point(size=5)</span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><p class="c5"><span style="overflow: hidden; display: inline-block; margin: 0.00px 0.00px; border: 0.00px solid #000000; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px); width: 624.00px; height: 312.00px;"><img alt="Amplicon-WGS.png" src="images/image00.png" style="width: 624.00px; height: 312.00px; margin-left: 0.00px; margin-top: 0.00px; transform: rotate(0.00rad) translateZ(0px); -webkit-transform: rotate(0.00rad) translateZ(0px);" title=""></span></p><p class="c3"><span></span></p><p class="c3"><span></span></p><h1 class="c5 c8"><a name="h.6fctr42y4zjd"></a><span>Example 2 - Shell script - Multiple alignment</span></h1><p class="c3"><span></span></p><p class="c5 c6"><span>This script uses the API directly to construct a multiple-sequence alignment of sequences that have similarity to a particular protein template.</span></p><p class="c2"><span></span></p><p class="c5"><span class="c0"># retrieve an aldehyde dehydrogenease protein sequence from M5NR</span></p><p class="c5"><span class="c0">ID=20f593463755f1080b867f55e368cef0</span></p><p class="c5"><span class="c0">curl &quot;</span><span class="c9 c0"><a class="c1" href="https://www.google.com/url?q=https://mail.anl.gov/owa/redir.aspx?SURL%3D4FiI1kFC2OkwKRp_3-a6JgwkDbbywdTe8ocEbMxe4Oz84l5PbOTSCGgAdAB0AHAAOgAvAC8AYQBwAGkALgBtAGUAdABhAGcAZQBuAG8AbQBpAGMAcwAuAGEAbgBsAC4AZwBvAHYALwBtADUAbgByAC8AbQBkADUALwAkAEkARAA_AHMAZQBxAHUAZQBuAGMAZQA9ADEAJgBmAG8AcgBtAGEAdAA9AGYAYQBzAHQAYQAmAHYAZQByAHMAaQBvAG4APQAxAA..%26URL%3Dhttp%253a%252f%252fapi.metagenomics.anl.gov%252fm5nr%252fmd5%252f%2524ID%253fsequence%253d1%2526format%253dfasta%2526version%253d1&amp;sa=D&amp;usg=AFQjCNFPc8PahH5VhHeMgTB8G2_R5DMQ3g">http://api.metagenomics.anl.gov/m5nr/md5/$ID?sequence=1&amp;format=fasta&amp;version=1</a></span><span class="c0">&quot; &gt; PYR-ref.fa</span></p><p class="c3"><span class="c0"></span></p><p class="c5"><span class="c0"># crop the reference protein sequence</span></p><p class="c5"><span class="c0">echo &quot;&gt;$ID-abbreviated&quot; &gt; PYR-ref-abbrev.fa</span></p><p class="c5"><span class="c0">cut -c 450-521 PYR-ref.fa | tail -n 1 &nbsp;&gt;&gt; PYR-ref-abbrev.fa</span></p><p class="c3"><span class="c0"></span></p><p class="c5"><span class="c0"># variables to find the things</span></p><p class="c5"><span class="c0">FILTER=1.5.1.12 &nbsp;# Delta-1-pyrroline-5-carboxylate dehydrogenase</span></p><p class="c5"><span class="c0">FILT=PYR</span></p><p class="c3"><span class="c0"></span></p><p class="c5"><span class="c0">curl &quot;</span><span class="c9 c0"><a class="c1" href="https://www.google.com/url?q=https://mail.anl.gov/owa/redir.aspx?SURL%3DodVRzPvajLm1IYFdKpEgD4w4AP2aDSZKa-0pmvxF8ST84l5PbOTSCGgAdAB0AHAAOgAvAC8AYQBwAGkALgBtAGUAdABhAGcAZQBuAG8AbQBpAGMAcwAuAGEAbgBsAC4AZwBvAHYALwAvAGEAbgBuAG8AdABhAHQAaQBvAG4ALwBzAGUAcQB1AGUAbgBjAGUALwA0ADUAMwA5ADUAMQA4AC4AMwA_AGUAdgBhAGwAdQBlAD0AMQAmAHQAeQBwAGUAPQBmAHUAbgBjAHQAaQBvAG4AJgBzAG8AdQByAGMAZQA9AFMAdQBiAHMAeQBzAHQAZQBtAHMAJgBmAGkAbAB0AGUAcgA9ACQARgBJAEwAVABFAFIA%26URL%3Dhttp%253a%252f%252fapi.metagenomics.anl.gov%252f%252fannotation%252fsequence%252f4539518.3%253fevalue%253d1%2526type%253dfunction%2526source%253dSubsystems%2526filter%253d%2524FILTER&amp;sa=D&amp;usg=AFQjCNEyIWxp-xz9nvTgX305YVG4QniPrA">http://api.metagenomics.anl.gov//annotation/sequence/4539518.3?evalue=1&amp;type=function&amp;source=Subsystems&amp;filter=$FILTER</a></span><span class="c0">&quot; | grep -v &#39;sequence id&#39; | head -n 20 &gt; $FILT-GP.txt</span></p><p class="c5"><span class="c0">curl &quot;</span><span class="c0 c9"><a class="c1" href="https://www.google.com/url?q=https://mail.anl.gov/owa/redir.aspx?SURL%3D_J79DdOBis3EbuXD8MlkpnY34a1YMviNuOAW1_1OLkb84l5PbOTSCGgAdAB0AHAAOgAvAC8AYQBwAGkALgBtAGUAdABhAGcAZQBuAG8AbQBpAGMAcwAuAGEAbgBsAC4AZwBvAHYALwAvAGEAbgBuAG8AdABhAHQAaQBvAG4ALwBzAGUAcQB1AGUAbgBjAGUALwA0ADUAMQA2ADMANgA2AC4AMwA_AGUAdgBhAGwAdQBlAD0AMQAmAHQAeQBwAGUAPQBmAHUAbgBjAHQAaQBvAG4AJgBzAG8AdQByAGMAZQA9AFMAdQBiAHMAeQBzAHQAZQBtAHMAJgBmAGkAbAB0AGUAcgA9ACQARgBJAEwAVABFAFIA%26URL%3Dhttp%253a%252f%252fapi.metagenomics.anl.gov%252f%252fannotation%252fsequence%252f4516366.3%253fevalue%253d1%2526type%253dfunction%2526source%253dSubsystems%2526filter%253d%2524FILTER&amp;sa=D&amp;usg=AFQjCNEVYMpDRJOfhV_73f8rTp1mi-dJcg">http://api.metagenomics.anl.gov//annotation/sequence/4516366.3?evalue=1&amp;type=function&amp;source=Subsystems&amp;filter=$FILTER</a></span><span class="c0">&quot; | grep -v &#39;sequence id&#39; | head -n 20 &nbsp;&gt; $FILT-BJ.txt</span></p><p class="c3"><span class="c0"></span></p><p class="c5"><span class="c0"># convert format of table to nucleotide fasta</span></p><p class="c5"><span class="c0">perl -nle &#39;@f=split;print &quot;&gt;$f[0] $f[1]\n$f[2]&quot;&#39; $FILT-GP.txt &nbsp;&gt; $FILT-GP.fa</span></p><p class="c5"><span class="c0">perl -nle &#39;@f=split;print &quot;&gt;$f[0] $f[1]\n$f[2]&quot;&#39; $FILT-BJ.txt &nbsp;&gt; $FILT-BJ.fa</span></p><p class="c3"><span class="c0"></span></p><p class="c5"><span class="c0"># predict translation of fragments</span></p><p class="c5"><span class="c0">run_FragGeneScan.pl -genome $FILT-GP.fa -out $FILT-GP -complete 0 -train complete</span></p><p class="c5"><span class="c0">run_FragGeneScan.pl -genome $FILT-BJ.fa -out $FILT-BJ -complete 0 -train complete</span></p><p class="c3"><span class="c0"></span></p><p class="c5"><span class="c0"># perform protein multiple sequence alignment</span></p><p class="c5"><span class="c0">cat PYR-ref-abbrev.fa $FILT-GP.faa $FILT-BJ.faa &gt; $FILT-foralignment.fa</span></p><p class="c5"><span class="c0">muscle -in $FILT-foralignment.fa -out $FILT-foralignment.aln.fasta</span></p></body></html>